src/main/java/frc/robot/subsystems/Vision/VisionIOReal.java

```diff
@@ -23,20 +29,20 @@ public class VisionIOReal implements VisionIO {
   public Transform3d robotToCamera;
 
   public VisionIOReal(VisionConstants constants) {
-    CAMERA_NAME = constants.cameraName();
-    camera = new PhotonCamera(CAMERA_NAME);
+    cameraName = constants.cameraName();
+    camera = new PhotonCamera(cameraName);
     robotToCamera = constants.robotToCamera();
+    cameraMatrixOpt = constants.CAMERA_MATRIX_OPT();
+    distCoeffsOpt = constants.DIST_COEFFS_OPT();
   }
 
   @Override
-  public void updateInputs(VisionIOInputs inputs, Pose3d pose) { // TODO
+  public void updateInputs(VisionIOInputs inputs, Pose3d pose) {
     var result = camera.getLatestResult();
     inputs.timestamp = result.getTimestampSeconds();
     inputs.latency = result.getLatencyMillis();
-    inputs.coprocPNPTargets = result.targets;
+    inputs.targets = result.targets;
     inputs.numTags = result.targets.size();
-    inputs.pose = pose;
+    inputs.coprocPNPPose = pose;
```

Comment:
dont just pass through the input pose here

Reply:
wait wdym

Reply:
`pose` is the input pose, or the previously estimated pose. you want coprocPNPPose to be the, well coprocPNPPose